lib/view/list_widget.dart

```diff
@@ -73,18 +73,28 @@ class _ListWidgetState extends State<ListWidget> {
   }
 
   Iterable<Widget> _buildItemWidgets({required Iterable<Item> fromItems}) =>
-      fromItems.map((item) => ItemWidget(item, onDelete: () async {
+      fromItems.map((item) => ItemWidget(item,
+          onDelete: () async {
             await listModel.remove(item);
             await refreshItems();
-          }, onEdited: () async {
-            try {
-              await listModel.update(item);
-              setState(() {});
-            } on ItemUpdateError catch (e) {
-              // TODO: handle item update error.
-              debugPrint(e.toString());
-            }
-          }));
+          },
+          onEdited: () => _onItemEdited(item)));
+
+  void _onItemEdited(Item item) async {
```

Comment:
This function is a serious smell. We shouldn't need to call `listModel.update` inside it, `listModel` should handle that automatically.